### PR TITLE
Add manifest functions: find_linecard, prop_supported

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
-Fix demo_vrf.pp to use these functions

### CODE REVIEW

File: lib/puppet/parser/functions/find_linecard.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    # Input : String, Linecard name, eg. 'N7K-F3'
+    # Output: Slot number of first match found
+    #
+    module Functions
+      newfunction(:find_linecard, type: :rvalue) do |args|
+        data = lookupvar('cisco')
+        return '' if data.nil?
+
+        pat = Regexp.new(args[0])
+        inv = data['inventory']
```
Comment (mikewiebe) [inline lib/puppet/parser/functions/find_linecard.rb]: Do you  also need a nil check for `data['inventory']`?

File: lib/puppet/parser/functions/find_linecard.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        match = inv.keys.select { |slot| inv[slot]['pid'].match(pat) }
```
Comment (mikewiebe) [inline lib/puppet/parser/functions/find_linecard.rb]: How are multiple matches treated here.  Does it just return the first match?

File: lib/puppet/parser/functions/find_linecard.rb (hunk 1 above)
Comment (chrisvanheuveln) [inline lib/puppet/parser/functions/find_linecard.rb]: Nope. `inv.keys` is an array and it's enumerable.

File: lib/puppet/parser/functions/find_linecard.rb (hunk 2 above)
Comment (chrisvanheuveln) [inline lib/puppet/parser/functions/find_linecard.rb]: First match. See comment on line 26.

**Comment by mikewiebe** [comment]
:+1:

**Pull request merged**